### PR TITLE
fix gac and slightly tweak build system

### PR DESCRIPTION
**Pull request by fingolfin**

### CODE REVIEW

File: cnf/Makegap.in [hunk 1]
```diff
@@ -683,3 +674,15 @@ gapmpi.o: @srcdir@/system.h @srcdir@/gasman.h @srcdir@/objects.h \
           @srcdir@/vars.h @srcdir@/stats.h @srcdir@/gapmpi.h 
 	$(CC) -I. $(MPI_CFLAGS) -DCONFIG_H $(CFLAGS) -o gapmpi.o -c @srcdir@/gapmpi.c
```
Comment (ChrisJefferson) [inline cnf/Makegap.in]: Why is there now two copies of this `Cygwin` code, both here and in `Makegap.bottom`?

Comment (fingolfin) [inline cnf/Makegap.in]: There always two copies of that code, I just moved one from Makefile to Makegap.bottom

Comment (fingolfin) [inline cnf/Makegap.in]: To expand on this: `Makegap.in` is a generated file, just like `configure.out`. We are intentionally violating the rule that generated files should not be in the repository (whether that violation is a good idea or not is another matter, and irrelevant for this PR). Both files are generated by `Makefile`. Part of what this PR does is to turn some of the rather opaque and hard to read / debug / change `echo` commands in `Makefile` into text in `Makegap.bottom` which is simply copied to `Makegap.in`. That makes it a bit easier to understand what's going on, and to tweak those build rules if necessary.

**Comment by ChrisJefferson** [comment]
I'm now happy with this. Someone else can merge it if they like, or I will in 24 hours or so.

**Pull request merged**